_includes/config.js.md

```diff
@@ -0,0 +1,413 @@
+
+Alternatively, you can specify the `bindDN` parameter. This will make The
+Lounge ignore `searchDN` options and assume that the user DN is always
+`<bindDN>,<primaryKey>=<username>`, where `<username>` is the user name
+provided in the log in request, and `<bindDN>` and `<primaryKey>` are
+provided by the configuration.
+
+The available keys for the `ldap` object are:
+
+- `enable`: when set to `false`, LDAP support is disabled and all other
+values are ignored.
+
+- `url`
+
+- `tlsOptions`: LDAP connection TLS options (only used if scheme is
+  `ldaps://`). It is an object whose values are Node.js' `tls.connect()`
+  options. It is set to `{}` by default.
+  For example, this option can be used in order to force the use of IPv6:
+  ```js
+  {
+    host: 'my::ip::v6',
+    servername: 'example.com'
+  }
+  ```
+
+- `baseDN`: LDAP base DN, alternative to `searchDN`. For example, set it
+  to `"ou=accounts,dc=example,dc=com"`.
+  It is not set by default, to use `searchDN` instead.
+- `primaryKey`: LDAP primary key. It is set to `"uid"` by default.
+
+- `searchDN`: LDAP search DN settings. This defines the procedure by
+  which The Lounge first looks for the user DN before authenticating them.
+  It is ignored if `baseDN` is specified. It is an object with the
+following keys:
+
+  - `rootDN`: This bind DN is used to query the server for the DN of
+    the user. This is supposed to be a system user that has access in
+    read-only to the DNs of the people that are allowed to log in.
+    It is set to `"cn=thelounge,ou=system-users,dc=example,dc=com"` by
+    default.
+
+  - `rootPassword`: Password of The Lounge LDAP system user.
+
+  - `ldapFilter`: it is set to `"(objectClass=person)(memberOf=ou=accounts,dc=example,dc=com)"`
+    by default.
+
+  - `base`: LDAP search base (search only within this node). It is set
+    to `"dc=example,dc=com"` by default.
+
+  - `scope`: LDAP search scope. It is set to `"sub"` by default.
+
+## Debugging settings
+
+The `debug` object contains several settings to enable debugging in The
+Lounge. Use them to learn more about an issue you are noticing but be aware
+this may produce more logging or may affect connection performance so it is
+not recommended to use them by default.
+
+All values in the `debug` object are set to `false`.
+
+### `debug.ircFramework`
+
+When set to true, this enables extra debugging output provided by
+[`irc-framework`](https://github.com/kiwiirc/irc-framework), the
+underlying IRC library for Node.js used by The Lounge.
+
+### `debug.raw`
+
+When set to `true`, this enables logging of raw IRC messages into each
+server window, displayed on the client.
+
+<!--
+DO NOT EDIT THIS FILE MANUALLY.
+Content for the following is generated by this script in the main repo:
+https://github.com/thelounge/thelounge/blob/master/scripts/generate-config-doc.js
+Last updated at Mon Sep 03 2018 23:52:23 GMT-0400 (EDT) by Jérémie Astori
```

Comment:
I think we should specify timestamp only once in the file (e.g. at the top), otherwise any update will bring 2 changed lines instead of 1.